src/modules/submissions.js

```diff
@@ -1985,8 +1981,29 @@
 		addFormSubmitHandler( handleDecline );
 	}
 
+	function addSignature( text ) {
+		text = text.trim();
+		if ( text.indexOf( '~~' + '~~' ) === -1 ) {
+			text += ' ~~' + '~~';
+		}
+		return text;
+	}
+
+	function previewComment() {
+		var commentText = $( '#commentText' ).val();
+		if ( commentText ) {
+			AFCH.api.parse( '{{AfC comment|' + addSignature( commentText ) + '}}', {
+				pst: true
+			} ).then( function ( html ) {
+				html = html.replace( /<script/g, '&lt;script' ); // just to be safe
+				$( '#commentPreview' ).html( html );
+			} );
+		}
+	}
+
 	function showCommentOptions() {
 		loadView( 'comment', {} );
+		$( '#commentText' ).on( 'keyup', mw.util.debounce( 500, previewComment ) );
```

Comment:
The use of debounce here causes the preview to be generated only if the user hasn't been typing anything for 500 milliseconds.